File: eHour-persistence/src/main/java/net/rrm/ehour/timesheet/dao/TimesheetCommentDaoHibernateImpl.java
/*
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 */

package net.rrm.ehour.timesheet.dao;

import net.rrm.ehour.dao.AbstractGenericDaoHibernateImpl;
import net.rrm.ehour.domain.TimesheetComment;
import net.rrm.ehour.domain.TimesheetCommentId;

import org.hibernate.Query;
import org.hibernate.Session;
import org.springframework.stereotype.Repository;

/**
 * CRUD on timesheetComment domain obj 
 **/
@Repository("timesheetCommentDao")
public class TimesheetCommentDaoHibernateImpl 
			extends AbstractGenericDaoHibernateImpl<TimesheetComment, TimesheetCommentId>
			implements TimesheetCommentDao
{
	public TimesheetCommentDaoHibernateImpl()
	{
		super(TimesheetComment.class);
	}

	public int deleteCommentsForUser(Integer userId)
	{
		Session session = getSession();
		Query query = session.getNamedQuery("TimesheetComment.deleteUserId");
		query.setParameter("userId", userId);
		
		return query.executeUpdate();
	}	
}

File: eHour-persistence/src/main/scala/net/rrm/ehour/persistence/timesheet/dao/TimesheetCommentDaoHibernateImpl.scala
package net.rrm.ehour.persistence.timesheet.dao

import java.util

import net.rrm.ehour.data.DateRange
import net.rrm.ehour.domain.{TimesheetComment, TimesheetCommentId}
import net.rrm.ehour.persistence.dao.AbstractGenericDaoHibernateImpl
import net.rrm.ehour.persistence.retry.ExponentialBackoffRetryPolicy
import org.springframework.stereotype.Repository

/**
 * CRUD on timesheetComment domain obj
 **/
@Repository("timesheetCommentDao")
class TimesheetCommentDaoHibernateImpl extends AbstractGenericDaoHibernateImpl[TimesheetCommentId, TimesheetComment](classOf[TimesheetComment]) with TimesheetCommentDao {
  override def deleteCommentsForUser(userId: Integer): Int = {
    val session = getSession
    val query = session.getNamedQuery("TimesheetComment.deleteUserId")
    query.setParameter("userId", userId)

    ExponentialBackoffRetryPolicy retry query.executeUpdate
  }

  override def findCommentBetween(dateRange: DateRange): util.List[TimesheetComment] = {
    val keys = List("dateStart", "dateEnd")
    val params = List(dateRange.getDateStart, dateRange.getDateEnd)
    findByNamedQuery("TimesheetComment.findCommentsIn", keys, params)
  }

  override def findCommentBetweenForUsers(users: util.List[Integer], dateRange: DateRange): util.List[TimesheetComment] = {
    val keys = List("dateStart", "dateEnd", "userIds")
    val params = List(dateRange.getDateStart, dateRange.getDateEnd, users)
    findByNamedQuery("TimesheetComment.findCommentsForUsersIn", keys, params)
  }
}